### PR TITLE
[3.0] Do not set exception_check if already set during analyzing pxd file

### PR DESCRIPTION
**Pull request by matusvalo**
Backport of #6124

### CODE REVIEW

**Review by da-woods — APPROVED** [review]

**Comment by matusvalo** [comment]
@da-woods pypy27 is failing, not understanding why. I pushed a workaround of CI to fix the issue with encoding...

**Comment by da-woods** [comment]
> pypy27 is failing, not understanding why. I pushed a workaround of CI to fix the issue with encoding to show the error message...

I've seen similar errors before and also not understood it. I'll remove my approval for now until I have a chance to look at the encoding properly

**Comment by matusvalo** [comment]
OK here we go this is error for pypy27:

```
Errors from shard 6:
======================================================================
FAIL: runTest (__main__.EndToEndTest)
[6] End-to-end legacy_implicit_noexcept_build
----------------------------------------------------------------------
Traceback (most recent call last):
  File "runtests.py", line 2102, in runTest
    ' '.join(command), self._try_decode(out[-1]), self._try_decode(err[-1])))
AssertionError: non-zero exit status, last output was:
'/opt/hostedtoolcache/PyPy/2.7.18/x64/bin/python -c import run'
-- stdout:('m2 1', False)
('m2 2', False)
('m2 1', True)

-- stderr:Traceback (most recent call last):
  File "<module>", line 1, in <module>
  File "run.py", line 25, in <module>
    check_exception_not_propagated(klass.m2, 2)
  File "run.py", line 18, in check_exception_not_propagated
    assert False, "ValueError raised"
AssertionError: ValueError raised
```

I have no idea why this happens only for pypy 2.7...

**Comment by da-woods** [comment]
@matusvalo The issue is https://github.com/pypy/pypy/issues/3185 - PyPy 2 just imports the .py file rather than the Cython-generated extension.

I think just skip the test on pypy2 (tests/pypy2_bugs). It isn't worth the time writing a workaround.

**Comment by matusvalo** [comment]
I will implement that in next days. Thank you @da-woods for your help.

File: runtests.py [hunk 1]
```diff
@@ -2047,12 +2047,18 @@ def tearDown(self):
...
-            return content
-        try:
-            return content.decode()
-        except UnicodeDecodeError:
-            return content.decode('iso-8859-1')
+        if IS_PY2:
+            if isinstance(content, bytes):
+                return content
+            else:
+                content.encode(errors='replace')
```
[the remark this answers is not in the document]
Comment (matusvalo) [reply runtests.py]: No this change has only debug purpose. It is removed from PR now...

**Review by da-woods — APPROVED** [review]

**Pull request merged**